### PR TITLE
Added Ellipsoid surface area

### PR DESCRIPTION
**Pull request by ErixenCruz**
Fixed `Ellipsoid.geodeticSurfaceNormal` dividing by zero when given origin as input.
Added `Ellipsoid.surfaceArea` for computing the surface area under given rectangle.

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thank you so much for the pull request @ErixenCruz! I noticed this is your first pull request and I wanted to say welcome to the Cesium community!

The [Pull Request Guidelines](https://github.com/CesiumGS/cesium/blob/master/CONTRIBUTING.md#pull-request-guidelines) is a handy reference for making sure your PR gets accepted quickly, so make sure to skim that.

* :x: Missing `CONTRIBUTORS.md` entry.
    * Please add yourself to the [contributors](https://github.com/CesiumGS/cesium/blob/ellipsoid-surfacearea/CONTRIBUTORS.md) file!
* :heavy_check_mark: Signed CLA found.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.
- [ ] Works (or fails gracefully) in IE11.

File: CHANGES.md [hunk 1]
```diff
@@ -8,10 +8,12 @@
 
 ##### Additions :tada:
 
+- Added `Ellipsoid.surfaceArea` for computing the surface area under given rectangle.
 - Added support for PolylineVolume in CZML [#8841](https://github.com/CesiumGS/cesium/pull/8841)
 
 ##### Fixes :wrench:
 
+- Fixed `Ellipsoid.geodeticSurfaceNormal` dividing by zero when given origin as input.
```
Comment (lilleyse) [inline CHANGES.md:16]:
```suggestion
- Fixed a divide-by-zero bug in `Ellipsoid.geodeticSurfaceNormal` when given the origin as input. `undefined` is returned instead. [#8986](https://github.com/CesiumGS/cesium/pull/8986)
```

File: CHANGES.md [hunk 2]
```diff
@@ -8,10 +8,12 @@
 
 ##### Additions :tada:
 
+- Added `Ellipsoid.surfaceArea` for computing the surface area under given rectangle.
```
Comment (lilleyse) [inline CHANGES.md:11]:
```suggestion
- Added `Ellipsoid.surfaceArea` for computing the approximate surface area of a rectangle on the surface of an ellipsoid.
```

File: Specs/Core/EllipsoidSpec.js [hunk 3]
```diff
@@ -3,6 +3,7 @@ import { Cartographic } from "../../Source/Cesium.js";
 import { Ellipsoid } from "../../Source/Cesium.js";
 import { Math as CesiumMath } from "../../Source/Cesium.js";
 import createPackableSpecs from "../createPackableSpecs.js";
+import { Rectangle } from "../../Source/Cesium.js";
```
Comment (lilleyse) [inline Specs/Core/EllipsoidSpec.js:6]: Just to keep things tidy, move this `import` above one line so it's next to the other `Cesium.js` imports.

File: Source/Core/Ellipsoid.js [hunk 4]
```diff
@@ -688,4 +693,112 @@ Ellipsoid.prototype.getSurfaceNormalIntersectionWithZAxis = function (
...
+ * @callback Ellipsoid~RealValuedScalarFunction
+ *
+ * @param {Number} x The value used to evaluate the function.
+ * @returns {Number} The value of the function at x.
+ *
+ * @private
+ */
+
+/**
+ * Computes an approximation of the surface area of a given section of the surface of an ellipsoid using
```
Comment (lilleyse) [inline Source/Core/Ellipsoid.js:758]:
```suggestion
 * Computes an approximation of the surface area of a rectangle on the surface of an ellipsoid using
```

File: Source/Core/Ellipsoid.js [hunk 5]
```diff
@@ -688,4 +693,112 @@ Ellipsoid.prototype.getSurfaceNormalIntersectionWithZAxis = function (
...
+ * Gauss-Legendre 10th order quadrature.
+ *
+ * @param {Rectangle} rectangle The rectangle to find the surface area of.
+ * @returns {Number} The approximate area of the section on the surface of this ellipsoid.
```
Comment (lilleyse) [inline Source/Core/Ellipsoid.js:762]:
```suggestion
 * @returns {Number} The approximate area of the rectangle on the surface of this ellipsoid.
```

File: Source/Core/Ellipsoid.js [hunk 6]
```diff
@@ -688,4 +693,112 @@ Ellipsoid.prototype.getSurfaceNormalIntersectionWithZAxis = function (
...
+ * @returns {Number} The value of the function at x.
+ *
+ * @private
+ */
+
+/**
+ * Computes an approximation of the surface area of a given section of the surface of an ellipsoid using
+ * Gauss-Legendre 10th order quadrature.
+ *
+ * @param {Rectangle} rectangle The rectangle to find the surface area of.
```
Comment (lilleyse) [inline Source/Core/Ellipsoid.js:761]:
```suggestion
 * @param {Rectangle} rectangle The rectangle used for computing the surface area.
```

**Comment by ErixenCruz** [comment]
Ok. That last commit should do it @lilleyse Thanks for the feedback.

**Comment by ErixenCruz** [comment]
Just kidding I missed the CHANGES.md changes but that last commit should do it.

**Comment by ErixenCruz** [comment]
I know I'm not supposed to force, but I did it to fix typo in commit message.

**Comment by lilleyse** [comment]
Thanks @ErixenCruz!

**Pull request merged**